coins/monero/src/wallet/mod.rs

```diff
@@ -121,6 +123,11 @@ impl ViewPair {
       .concat(),
     ))
   }
+
+  /// Returns an address with the provided metadata.
+  pub fn address(&self, meta: AddressMeta<MoneroAddressBytes>) -> MoneroAddress {
+    MoneroAddress::new(meta, self.spend, self.view.deref() * &ED25519_BASEPOINT_TABLE)
```

Comment:
This isn't correct when the metadata specifies a subaddress.

Since AddressMeta doesn't include the subaddress index, there's no way to have a safe function with this API. It'd have to take payment ID/subaddress/guaranteed as individual arguments and then build a derivation/meta from that.